content/components/only-manipulate-the-dom-via-the-renderer.md

```diff
@@ -0,0 +1,50 @@
+---
+title: only manipulate the DOM via the Renderer
+author: 
+  name: Billy Lando (@billyjov)
+  url: https://github.com/billyjov
+---
+
+# Problem 
+
+According to the angular documentation, relying on direct DOM access creates tight coupling between your application and rendering layers which will make it impossible to separate the two and deploy your application into a web worker. Permitting direct access to the DOM can make your application more vulnerable to **XSS** attacks.
+
+# Solution 
+
+Always use as much as possible the `Renderer2` to manipulate the DOM. It provides API that can safely be used even when direct access to native elements is not supported.
```

Comment:
`Always use as much as possible the 'Renderer2' to manipulate the DOM.` => `Always try to prefer the 'Renderer2' for DOM manipulations.`

`It provides API that can...` => `It provides an API that can...`